src/main/java/apoc/util/DateParseUtil.java

```diff
@@ -0,0 +1,73 @@
+package apoc.util;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.reflect.InvocationTargetException;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.TemporalAccessor;
+import java.util.HashMap;
+import java.util.Map;
+
+import static apoc.util.Util.getFormat;
+
+public class DateParseUtil {
+
+    public static Map<Class<? extends TemporalAccessor>, MethodHandle> parseDateMap = new HashMap<>();
```

Comment:
and probably using a CHM for concurrent access.

Reply:
ok